sdk/samples/iot/pnp/pnp_protocol.c

```diff
@@ -194,43 +176,47 @@ void pnp_parse_command_name(
   }
 }
 
-az_result pnp_build_reported_property(
+void pnp_build_reported_property(
     az_span json_buffer,
     az_span component_name,
     az_span property_name,
     pnp_append_property_callback append_callback,
     void* context,
     az_span* out_span)
 {
+  char const* log = "Failed to build `%.*s` reported property";
```

Comment:
I see you have this pattern in multiple places, so I would suggestthe following for all occurrences:
1. Rename `log` to something like `error_msg`. But if that would result in autoformatting giving us too many line breaks, leave it as is, or use something shorter. `log` is not that bad, after all, I just think that `error_msg` is more descriptive. Up to you.
2. I would declare it as `char const* const`, so that when I am reading the code, I know you don't reassign it, so I shouldn't expect to track it being changed, as  read the code. Any place within the function, all the same meaning. But: for the SDK code I would definitely ask you to do that; here, in the sample, I leave it up to you, whether you think `char const* const` is too much of a complication for the majority of the audience, or not. For me, it certainly would be helpful, even if I wasn't one of the SDK devs :)

Reply:
I agree `error_msg` is much more descriptive of what it actually is.  The reason I went with `log` is because it was shorter.  Some of the lines are right on the cusp of having to use a second line if I go with anything much longer than `log`.  Intent was to save lines, making it cleaner to read. I'm split on it.

* const is definitely more descriptive.  I'll update those. Thanks.